frida_tools/tracer.py

```diff
@@ -918,6 +926,29 @@ def _sync_handlers(self, change_id):
                 self._notify_update(function, new_handler, handler_file)
 
 
+class OutputFile(object):
+    def __init__(self, filename, truncate=False):
+        self.filename = filename
+        self._truncate = truncate
+        self._fd = None
+
+    def __enter__(self):
+        if self.filename:
+            self._fd = codecs.open(self.filename, 'ab', 'utf-8')
+            if self._truncate:
+                self._fd.truncate(0)
+            return self
+        return None
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self._fd is not None:
+            self._fd.close()
```

Comment:
This code should probably be moved to a `close()` method that we call from here. That way this object can be used both as a scope-local context manager and as a long-lived instance.

Reply:
yes, the enter/exit scope magic is pretty messy. i have just removed that

Reply:
👍